nibabel/brainvoyager/bv.py

```diff
@@ -0,0 +1,957 @@
+        """Guess the dimensions of the framing cube.
+
+        Guess the dimensions of the framing cube that constitutes the
+        coordinate system boundaries for the bounding box.
+
+        For most BV file formats this need to be guessed from
+        x_end, y_end, and z_end in the header.
+        """
+        # then start guessing...
+        hdr = self._hdr_dict
+        # get the ends of the bounding box (highest values in each dimension)
+        x = hdr['x_end']
+        y = hdr['y_end']
+        z = hdr['z_end']
+
+        # compare with possible framing cubes
+        for fc in [256, 384, 512, 768, 1024]:
+            if any([d > fc for d in (x, y, z)]):
+                continue
+            else:
+                return fc, fc, fc
+
+    def get_framing_cube(self):
+        """Get the dimensions of the framing cube.
+
+        Get the dimensions of the framing cube that constitutes the
+        coordinate system boundaries for the bounding box.
+        For most BV file formats this need to be guessed from
+        x_end, y_end, and z_end in the header.
+        """
+        return self._framing_cube
+
+    def set_framing_cube(self, fc):
+        """Set the dimensions of the framing cube.
+
+        Set the dimensions of the framing cube that constitutes the
+        coordinate system boundaries for the bounding box
+        For most BV file formats this need to be guessed from
+        x_end, y_end, and z_end in the header.
+        Use this if you know about the framing cube for the BV file.
+        """
+        self._framing_cube = fc
+
+    def get_bbox_center(self):
+        """Get the center coordinate of the bounding box.
+
+        Get the center coordinate of the bounding box with respect to the
+        framing cube.
+        """
+        hdr = self._hdr_dict
+        x = (hdr['x_start'] +
+             ((hdr['x_end'] - hdr['x_start']) / 2))
+        y = (hdr['y_start'] +
+             ((hdr['y_end'] - hdr['y_start']) / 2))
+        z = (hdr['z_start'] +
+             ((hdr['z_end'] - hdr['z_start']) / 2))
+        return z, y, x
+
+    def get_zooms(self):
+        shape = self.get_data_shape()
+        return tuple(float(self._hdr_dict['resolution'])
+                     for d in shape[0:3])
+
+    def set_zooms(self, zooms):
+        if type(zooms) == int:
+            self._hdr_dict['resolution'] = zooms
+        else:
+            if any([zooms[i] != zooms[i + 1] for i in range(len(zooms) - 1)]):
+                raise BvError('Zooms for all dimensions must be equal!')
+            else:
+                self._hdr_dict['resolution'] = int(zooms[0])
+
+    def as_analyze_map(self):
+        raise NotImplementedError
+
+    def set_data_offset(self, offset):
+        """Set offset into data file to read data."""
+        self._data_offset = offset
+
+    def get_data_offset(self):
+        """Return offset into data file to read data."""
+        self.set_data_offset(calc_BV_header_size(
```

Comment:
Your call.  You can make `set_data_offset` raise an error if you like (parrec.py does this) - or you can let people do a `set` if they want, hoping they understand the risk.

Do people tend to store stuff in BV files after the header and before the data?  That's not uncommon in Analyze, but very rare for Niftis for example.  If it's rare, maybe raising an error is better.

By the way, you're using this in the header constructor, so I guess the `offset` in the header constructor is getting thrown away, in fact.  Maybe you can remove it from the header parameter list?